theme/templates/resource-landing-page/modals.html

```diff
@@ -0,0 +1,773 @@
+                <h4 class="modal-title" id="myModalLabel">Delete metadata element</h4></div>
+            <div class="modal-body"><strong>Are you sure you want to delete this metadata element?</strong>
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>
+                <a type="button" class="btn btn-danger"
+                   href="/hsapi/_internal/{{ cm.short_id }}/source/{{ source.id }}/delete-metadata/">Delete</a>
+            </div>
+        </div>
+    </div>
+</div>
+
+<div class="modal fade in" id="edit-source-dialog{{ source.id }}" tabindex="-1" role="dialog" aria-labelledby="myModalLabel"
+     aria-hidden="false">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <form id="id_source_{{ source.id }}" action="/hsapi/_internal/{{ cm.short_id }}/source/{{ source.id }}/update-metadata/"
+                  method="POST" enctype="multipart/form-data">
+                {% csrf_token %}
+                <input name="resource-mode" type="hidden" value="edit">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
+                    <h4 class="modal-title" id="myModalLabel"> Edit Source </h4></div>
+                <div class="modal-body">
+                    {% csrf_token %}
+                    <div class="form-group">
+                        {% csrf_token %}
+                        <fieldset>
+                            <legend>Source</legend>
+                            <div id="div_id_derived_from" class="control-group">
+                                <label for="id_derived_from" class="control-label requiredField">
+                                Derived from<span class="asteriskField">*</span></label>
+
+                                <div class="controls"><input
+                                        class="form-control input-sm textinput textInput"
+                                        id="id_source-{{ forloop.counter0 }}-derived_from" maxlength="300" name="source-{{ forloop.counter0 }}-derived_from"
+                                        type="text" value="{{ source.derived_from }}"></div>
+                            </div>
+                        </fieldset>
+                    </div>
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+                    <button type="submit" class="btn btn-primary">Save changes</button>
+                </div>
+            </form>
+        </div>
+    </div>
+</div>
+{% endfor %}
+
+{% for relation in relation_formset.initial %}
+<div class="modal fade in" id="delete-relation-element-dialog{{ relation.id }}" tabindex="-1" role="dialog"
```

Comment:
I think that would be better. The reason is that any work done in genericresource.html while this PR exists has to be manually included to this work.